pubs/apis.py

```diff
@@ -25,3 +27,33 @@ def isbn2bibtex(isbn):
     citation = soup.find("textarea").text
 
     return citation
+
+
+def arxiv2bibtex(arxiv_id):
+    """Return a bibtex string of metadata from an arXiv ID"""
+
+    url = 'https://export.arxiv.org/api/query?id_list=' + arxiv_id
+    r = requests.get(url)
+    feed = feedparser.parse(r.text)
+    entry = feed.entries[0]
+
+    if 'title' not in entry:
+        ui = get_ui()
+        ui.error('malformed arXiv ID: {}'.format(arxiv_id))
```

Comment:
In general, we try to keep the calls to UI errors outside of the helpers functions and mostly at the command layer (for instance this might not be an error but only a warning, which can only be decided at the command level). 

Here I would suggest raising a dedicated exception. You may reuse the same exception for various API issues and set the correct error message here, then call something like `ui.error(ustr(e))` when the exception is caught in the command code.

Reply:
This makes sense.